src/cdi_CPEloss/Tabular_CP_Eloss.cc

```diff
@@ -0,0 +1,246 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!                                                                              
+ * \file   cdi_CPEloss/Tabular_CP_Eloss.cc                                     
+ * \author Ben R. Ryan                                                        
+ * \date   2019 Nov 4                                             
+ * \brief  Tabular_CP_Eloss member definitions.                                 
+ * \note   Copyright (C) 2016-2019 Triad National Security, LLC.                 
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include "Tabular_CP_Eloss.hh"
+
+using std::stod;
+using std::stoi;
+
+namespace rtt_cdi_cpeloss {
+
+//---------------------------------------------------------------------------//
+// CONSTRUCTORS
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for an analytic tabular eloss model. 
+ *
+ * This constructor builds an eloss model defined by the
+ * rtt_cdi_cpeloss::Tabular_Eloss_Model derived class argument.
+ *
+ * The path to an eloss datafile is passed to the constructor,
+ * which opens and parses the file. The file format is the usual
+ * LANL format for stopping powers. 
+ *
+ * \param filename_in path to eloss file
+ * \param target_zaid_in target particle zaid
+ * \param projectile_zaid_in transporting particle zaid
+ */
+Tabular_CP_Eloss::Tabular_CP_Eloss(std::string filename_in,
+                                   rtt_cdi::CParticle &target_in,
+                                   rtt_cdi::CParticle &projectile_in)
+    : filename(filename_in), rtt_cdi::CPEloss(target_in, projectile_in) {
+
+  model_type = rtt_cdi::CPModelType::TABULAR_ETYPE;
+
+  file.open(filename);
+  if (!file.is_open()) {
+    std::cout << "Eloss file " << filename << " could not be opened!"
+              << std::endl;
+    exit(-1);
+  }
+
+  std::string line;
+  std::vector<std::string> line_entries;
+  int nlines;
+  int max_entries =
+      6; // This is a statement about the file format, maximum of six entries for row.
+
+  read_line(); // ZAID
+  int32_t projectile_zaid_file = stoi(line_entries[0]);
```

Comment:
I think you meant to set line_entries = read_line() on line 55 -- otherwise 56 will segfault! :O

Reply:
🤦‍♂️